internal/generated/clientset/versioned/clientset.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2018 Heptio
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package versioned
+
+import (
+	glog "github.com/golang/glog"
+	contourv1alpha1 "github.com/heptio/contour/internal/generated/clientset/versioned/typed/contour/v1alpha1"
+	discovery "k8s.io/client-go/discovery"
+	rest "k8s.io/client-go/rest"
+	flowcontrol "k8s.io/client-go/util/flowcontrol"
+)
+
+type Interface interface {
+	Discovery() discovery.DiscoveryInterface
+	ContourV1alpha1() contourv1alpha1.ContourV1alpha1Interface
+	// Deprecated: please explicitly pick a version if possible.
+	Contour() contourv1alpha1.ContourV1alpha1Interface
+}
+
+// Clientset contains the clients for groups. Each group has exactly one
+// version included in a Clientset.
+type Clientset struct {
+	*discovery.DiscoveryClient
+	contourV1alpha1 *contourv1alpha1.ContourV1alpha1Client
+}
+
+// ContourV1alpha1 retrieves the ContourV1alpha1Client
+func (c *Clientset) ContourV1alpha1() contourv1alpha1.ContourV1alpha1Interface {
+	return c.contourV1alpha1
+}
+
+// Deprecated: Contour retrieves the default version of ContourClient.
+// Please explicitly pick a version.
+func (c *Clientset) Contour() contourv1alpha1.ContourV1alpha1Interface {
+	return c.contourV1alpha1
+}
+
+// Discovery retrieves the DiscoveryClient
+func (c *Clientset) Discovery() discovery.DiscoveryInterface {
+	if c == nil {
+		return nil
+	}
+	return c.DiscoveryClient
+}
+
+// NewForConfig creates a new Clientset for the given config.
+func NewForConfig(c *rest.Config) (*Clientset, error) {
+	configShallowCopy := *c
+	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS > 0 {
+		configShallowCopy.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(configShallowCopy.QPS, configShallowCopy.Burst)
+	}
+	var cs Clientset
+	var err error
+	cs.contourV1alpha1, err = contourv1alpha1.NewForConfig(&configShallowCopy)
+	if err != nil {
+		return nil, err
+	}
+
+	cs.DiscoveryClient, err = discovery.NewDiscoveryClientForConfig(&configShallowCopy)
+	if err != nil {
+		glog.Errorf("failed to create the DiscoveryClient: %v", err)
```

Comment:
rather than logging here, as the layer above is going to log this; what about
return nil, fmt.Errorf("failed to create discoveryclient: %v", err)

You could also use github.com/pkg/errors, but I don't think it's in the dependency tree at the moment and I don't think anyone is going to want to recover the original error -- it's just going to end up hitting check() at the top level.

Reply:
this is autogenerated, not sure how much control we have over how it's handled